prov/netdir/src/netdir_init.c

```diff
@@ -104,22 +104,22 @@ static int ofi_nd_adapter_cb(const ND2_ADAPTER_INFO *adapter, const char *name)
 	info->tx_attr->mode = FI_CONTEXT;
 	info->tx_attr->comp_order = FI_ORDER_STRICT;
 	info->tx_attr->inject_size = (size_t)gl_data.inline_thr;
-	info->tx_attr->size = (size_t)adapter->MaxTransferLength;
+	info->tx_attr->size = (size_t)adapter->MaxInitiatorQueueDepth;
 	/* TODO: if optimization will be needed, we can use adapter->MaxInitiatorSge,
 	 * and use ND SGE to send/write iovecs */
-	info->tx_attr->iov_limit = ND_MSG_IOV_LIMIT;
-	info->tx_attr->rma_iov_limit = ND_MSG_IOV_LIMIT;
+	info->tx_attr->iov_limit = (size_t)adapter->MaxInitiatorSge;
+	info->tx_attr->rma_iov_limit = (size_t)adapter->MaxInitiatorSge;
 	info->tx_attr->op_flags = OFI_ND_TX_OP_FLAGS;
 	info->tx_attr->msg_order = OFI_ND_MSG_ORDER;
 
 	info->rx_attr->caps = FI_MSG | FI_RECV;
 	info->rx_attr->mode = FI_CONTEXT;
 	info->rx_attr->comp_order = FI_ORDER_STRICT;
 	info->rx_attr->total_buffered_recv = 0;
-	info->rx_attr->size = (size_t)adapter->MaxTransferLength;
+	info->rx_attr->size = (size_t)adapter->MaxReceiveQueueDepth;
 	/* TODO: if optimization will be needed, we can use adapter->MaxInitiatorSge,
 	 * and use ND SGE to recv iovecs */
-	info->rx_attr->iov_limit = ND_MSG_IOV_LIMIT;
+	info->rx_attr->iov_limit = (size_t)adapter->MaxReceiveSge;
```

Comment:
They way other providers handle this is that a default value is returned, rather than the supported max.  Many apps simply take the returned values and hand them right back.  And the max values are almost never ideal.

See how the tcp provider handles this using an alter_info callback from the core util getinfo code.  The callback is specified here:

https://github.com/ofiwg/libfabric/blob/main/prov/tcp/src/tcpx_attr.c#L187

which invokes this function (right above it):

https://github.com/ofiwg/libfabric/blob/main/prov/tcp/src/tcpx_attr.c#L171

Reply:
Made an attempt to provide defaults of 8 for iov_limit and 384 for size, which are close to the verbs defaults and also match our use case.